src/compute-types/src/plan/render_plan.rs

```diff
@@ -475,34 +507,50 @@ impl<T> From<Plan<T>> for FlatPlan<T> {
                     input_key,
                     input_mfp,
                 } => {
-                    let node = ArrangeBy {
+                    let expr = ArrangeBy {
                         input: input.lir_id,
                         forms,
                         input_key,
                         input_mfp,
                     };
-                    insert_node(lir_id, parent, node, nesting);
+                    insert_node(lir_id, parent, expr, nesting);
 
                     todo.push((*input, Some(lir_id), nesting.saturating_add(1)));
                 }
+                PlanNode::Let { .. } | PlanNode::LetRec { .. } => return Err(()),
```

Comment:
This error seems to be only way we can fail the `try_from`... and the only place we seem to call the `try_from`, we just `expect` or `unwrap` the answer. Why not use `From` and and `unreachable!` at that point?

Reply:
I see what you're saying---I don't have a strong opinion either. Might as well leave it.

I suppose the graceful failure here is to just not end up issuing _this_ query rather than crashing the cluster, but it would take a fair bit of plumbing that error back to do so. Until that plumbing happens, somebody _somewhere_ will have to panic if this invariant isn't correctly met, short of having something like `NormalizeLets` baked into lowering.